### PR TITLE
Implement WithVoidReturnType

### LINKED ISSUE #1272: Add WithVoidReturnType method
As mentioned a few times in the past, we should have a `WithVoidReturnType` method that mirrors `WithNonVoidReturnType`.

### PR DESCRIPTION
**Pull request by thomaslevesque**
Fixes #1272

### CODE REVIEW

File: src/FakeItEasy/Configuration/IAnyCallConfigurationWithNoReturnTypeSpecified.cs [hunk 1]
```diff
@@ -21,5 +21,11 @@ public interface IAnyCallConfigurationWithNoReturnTypeSpecified
         /// </summary>
         /// <returns>A configuration object.</returns>
         IAnyCallConfigurationWithReturnTypeSpecified<object> WithNonVoidReturnType();
+
+        /// <summary>
+        /// Matches calls that have void return type.
+        /// </summary>
+        /// <returns>A configuratino object.</returns>
```
Comment (blairconrad) [inline src/FakeItEasy/Configuration/IAnyCallConfigurationWithNoReturnTypeSpecified.cs]: "configuration"

File: docs/specifying-a-call-to-configure.md [hunk 2]
```diff
@@ -52,6 +52,9 @@ the fake to `A.CallTo` to refer to any method on the fake:
 ```csharp
 A.CallTo(fakeShop).Throws(new Exception());
 
+// Or limit the calls to void methods
+A.CallTo(fakeShop).WithVoidReturnType().Throws("sugar overflow");
```
Comment (blairconrad) [inline docs/specifying-a-call-to-configure.md]: 1. nice exception

File: src/FakeItEasy/Configuration/IAnyCallConfigurationWithVoidReturnType.cs [hunk 3]
```diff
@@ -0,0 +1,11 @@
+namespace FakeItEasy.Configuration
+{
+    /// <summary>
+    /// Aggregate of IVoidArgumentValidationConfiguration&lt;T&gt; and IWhereConfiguration&lt;IAnyCallConfigurationWithVoidReturnType&gt;.
```
Comment (blairconrad) [inline src/FakeItEasy/Configuration/IAnyCallConfigurationWithVoidReturnType.cs]: `IVoidArgumentValidationConfiguration` is not generic - strike the `&lt;T&gt;`.

**Comment by thomaslevesque** [comment]
@blairconrad thanks for the review. What about the breaking/non-breaking question I mentioned in the issue? If we consider this to be breaking, merging it now means the next version will be 5.0.0. (IMO we shouldn't consider this a breaking change; it's true that it changes an interface, but I'm willing to bet that not a single user will be affected)

File: src/FakeItEasy/Configuration/IAnyCallConfigurationWithNoReturnTypeSpecified.cs (hunk 1 above)
Comment (thomaslevesque) [reply src/FakeItEasy/Configuration/IAnyCallConfigurationWithNoReturnTypeSpecified.cs]: amended

**Comment by blairconrad** [comment]
@thomaslevesque, I was stalling with the easy review before addressing the breaking/non-breaking question. I will respond in the issue.

File: src/FakeItEasy/Configuration/IAnyCallConfigurationWithVoidReturnType.cs (hunk 3 above)
Comment (blairconrad) [reply src/FakeItEasy/Configuration/IAnyCallConfigurationWithVoidReturnType.cs]: @thomaslevesque, did this comment get lost in the excitement?

Comment (thomaslevesque) [reply src/FakeItEasy/Configuration/IAnyCallConfigurationWithVoidReturnType.cs]: Yes ! I'll fix it

Comment (thomaslevesque) [reply src/FakeItEasy/Configuration/IAnyCallConfigurationWithVoidReturnType.cs]: Fixed

**Comment by blairconrad** [comment]
Thanks, @thomaslevesque!

**Pull request merged**